xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/Registration.xml

```diff
@@ -763,4 +791,100 @@
+      <customClass/>
+      <customMapping/>
+      <defaultViewSheet/>
+      <defaultEditSheet/>
+      <defaultWeb/>
+      <nameField/>
+      <validationScript/>
+      <content>
+        <disabled>0</disabled>
+        <name>content</name>
+        <number>3</number>
+        <prettyName>Extension Content</prettyName>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </content>
+      <extensionPointId>
+        <disabled>0</disabled>
+        <name>extensionPointId</name>
+        <number>1</number>
+        <prettyName>Extension Point ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </extensionPointId>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Extension ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parameters>
+        <disabled>0</disabled>
+        <name>parameters</name>
+        <number>4</number>
+        <prettyName>Extension Parameters</prettyName>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </parameters>
+      <scope>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>scope</name>
+        <number>5</number>
+        <prettyName>Extension Scope</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>wiki=Current Wiki|user=Current User|global=Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </scope>
+    </class>
+    <property>
+      <content>{{velocity}}{{html clean="false"}}
+#if ($xcontext.user == 'XWiki.XWikiGuest' &amp;&amp; !$xcontext.inactiveUserReference)
+&lt;li&gt;
+  &lt;a href="$xwiki.getURL('XWiki.XWikiLogin', 'login', "xredirect=$escapetool.url($xwiki.relativeRequestURL)&amp;loginLink=1")" id="tmLogin" rel="nofollow"&gt;$escapetool.xml($services.localization.render('login'))&lt;/a&gt;
+&lt;/li&gt;
+#end
+#if ($xcontext.user == 'XWiki.XWikiGuest' &amp;&amp; !$xcontext.inactiveUserReference &amp;&amp; $xwiki.hasAccessLevel('register', 'XWiki.XWikiPreferences'))
+&lt;li&gt;
+&lt;a href="$xwiki.getURL('XWiki.XWikiRegister', 'register', "xredirect=$escapetool.url($xwiki.relativeRequestURL)")" id="tmRegister" rel="nofollow"&gt;$escapetool.xml($services.localization.render('register'))&lt;/a&gt;
+&lt;/li&gt;
```

Comment:
I'm wondering the impact of this for private wikis, you could have a wiki entirely private in which registration is still possible no? In which case here we're loosing seeing the register link at all, isn't it?

Reply:
If the wiki is entirely private, the user would not be able to see any of the buttons of the top right navigation bar anyways.
I think it's not that much of an issue.

Moreover, as far as I know, private wikis did not use this way to get new users:
* either invitation of new users
* or the admin creates the accounts and shares the logins with the actual users.

Maybe it'd be an issue for semi private?
Even in the case the Registration page is public, there's other ways to access it. 

Let me know if you find a use case where this is wrong.

Thanks,
LC

Reply:
Well if in XWiki 15.10.13 I set the rights to prevent unregistered users to view pages, but yet allow guest users to register I obtain this as guest: 
![image](https://github.com/user-attachments/assets/2e0d6371-a8d9-4bd5-986f-c3e35425f064)

Do you get the same with your changes?

Reply:
## Test conditions
I tried to reproduce what you described:
![image](https://github.com/user-attachments/assets/6366cb4b-f00b-4103-a097-9dacf504a005)
## Result
Trying to go anywhere while not registered prompts me with this page:
![image](https://github.com/user-attachments/assets/29a970fc-8ee6-41cc-a008-6b070585fefa)
(the drawer is similar to the one you sent, minus the two links)
Clicking on the register link displays the register form as expected :+1: 
![image](https://github.com/user-attachments/assets/f73b2d93-0a59-4042-abf4-d5c0f3f54e1d)

Reply:
Okay, I tried to find back where the set of conditions to display the buttons come from. 

It seems that it's just a factorization of all the conditions that were needed to display the links before. 
https://github.com/xwiki/xwiki-platform/pull/3155/files#diff-97052b73aa6732071455b709190c5a5e65729555d3587e57882b79ea2f57e12dL38-L58

Unless my boolean factorization is wrong, this should be displayed in the same conditions as the drawer links and not be a regression on feature.
The original code is not factorized like this because there are other moving parts in the template that are only displayed on a specific set of conditions.

Reply:
Ok thanks for checking this.